itests/kit/node_unmanaged.go

```diff
@@ -1474,20 +1474,20 @@ func (tm *TestUnmanagedMiner) WaitTillActivatedAndAssertPower(sectors []abi.Sect
 	for _, sectorNumber := range sectors {
 		tm.WaitTillPost(sectorNumber)
 
-		// Fetch on-chain sector properties
-		head, err := tm.FullNode.ChainHead(tm.ctx)
-		req.NoError(err)
-
-		soi, err := tm.FullNode.StateSectorGetInfo(tm.ctx, tm.ActorAddr, sectorNumber, head.Key())
-		req.NoError(err)
-		tm.log("SectorOnChainInfo %d: %+v", sectorNumber, soi)
-
-		_ = tm.FullNode.WaitTillChain(tm.ctx, HeightAtLeast(head.Height()+5))
-
-		if !tm.mockProofs {
+		if !tm.mockProofs { // else it would pass, which we don't want
+			// if the sector is in the current or previous deadline, we can't dispute the PoSt
+			di, err := tm.FullNode.StateMinerProvingDeadline(tm.ctx, tm.ActorAddr, types.EmptyTSK)
+			req.NoError(err)
+			sl, err := tm.FullNode.StateSectorPartition(tm.ctx, tm.ActorAddr, sectorNumber, types.EmptyTSK)
+			req.NoError(err)
+			if sl.Deadline == CurrentDeadlineIndex(di) {
+				tm.log("In current proving deadline for sector %d, waiting for the next deadline to dispute PoSt", sectorNumber)
```

Comment:
@rvagg Also need to ensure it's not in the previous deadline right ?

Reply:
no, there's 2 rules: proving deadline for the sector is closed, and we haven't had more than 2 days elapsed: https://github.com/filecoin-project/builtin-actors/blob/master/actors/miner/src/deadlines.rs#L86-L106